gapis/capture/encoder.go

```diff
@@ -11,23 +12,31 @@ import (
 	"github.com/google/gapid/gapis/database"
 )
 
+var mutex sync.Mutex
+
 type encoder struct {
 	c      *Capture
 	w      *pack.Writer
 	cmdIDs map[api.Cmd]uint64
-	seen   map[id.ID]bool
+	resIDs map[id.ID]int64
 }
 
 func newEncoder(c *Capture, w *pack.Writer) *encoder {
 	return &encoder{
 		c:      c,
 		w:      w,
 		cmdIDs: map[api.Cmd]uint64{},
-		seen:   map[id.ID]bool{},
+		resIDs: map[id.ID]int64{id.ID{}: 0},
```

Comment:
Nothing uses the resource 0, but I just really wanted to reserve 0 to mean null for robustness.

Say, if we ever encoded extra with empty id.ID in it, it would translate to 0.

Reply:
Okay, SGTM.